### PR TITLE
QPUDevice: Skip generate_samples to avoid executing a circuit twice

### PR DESCRIPTION
**Pull request by AlbertMitjans**
**Context**:
When running the following code, the circuit is executed twice: once by the `Device.generate_samples` method and another one by the `QPUDevice.expval` method, but only the results from the last method are used.
```python
dev = qml.device("forest.qpu", device="4q-qvm", shots=10000, load_qc=False, parametric_compilation=False)

@qml.qnode(dev)
def circuit():
    return qml.counts(qml.PauliZ(0))

print(circuit())
```
**Issue**: https://github.com/PennyLaneAI/pennylane-forest/issues/45
**Description of the change**: Skip `Device.generate_samples` when executing a circuit with `expval` and `parametric_compilation=False`
**Possible drawbacks**: We are overriding the `Device.execute` method, which is not recommended.

### CODE REVIEW

File: pennylane_forest/qpu.py [hunk 1]
```diff
@@ -139,11 +141,7 @@ def expval(self, observable, shot_range=None, bin_size=None):
                 pauli_obs = sZ(wire)
 
             # Multi-qubit observable
-            elif (
-                len(device_wires) > 1
-                and isinstance(observable, Tensor)
-                and not self.parametric_compilation
```
Comment (antalszava) [inline pennylane_forest/qpu.py:145]: Why was this changed? This seems to allow parametric compilation with op estimation, do they work well together in this plugin?

**Review by antalszava — COMMENTED** [review]
:100: I had one question. Don't forget to add an entry to the changelog. It likely would be required for #107 (and may have been skipped in the past :/).

File: pennylane_forest/qpu.py (hunk 1 above)
Comment (AlbertMitjans) [reply pennylane_forest/qpu.py:145]: We already check that `self.parametric_compilation` is False at the beginning of this method. This check here did nothing.

Comment (AlbertMitjans) [reply pennylane_forest/qpu.py:145]: I don't think they work well together, so we still disable operator estimation when `parametric_compilation = True`

**Comment by AlbertMitjans** [comment]
> Don't forget to add an entry to the changelog.

Added changelog entry.

> It likely would be required for https://github.com/PennyLaneAI/pennylane-forest/pull/107 (and may have been skipped in the past :/).

Oops my fault. I will tell the people of Rigetti to create a PR adding the changelog.

File: pennylane_forest/qpu.py (hunk 1 above)
Comment (antalszava) [reply pennylane_forest/qpu.py:145]: Good catch! Thanks for explaining.

**Review by antalszava — APPROVED** [review]
Looks good :+1:
